pkg/build/controller/build/build_controller.go

```diff
@@ -2333,6 +2333,46 @@ func isValidTransition(from, to buildv1.BuildPhase) bool {
 	return true
 }
 
+func checkLogSnippetsForBadBuildPhase(build *buildv1.Build, update *buildUpdate) bool {
```

Comment:
function name seems odd...  doesn't look at logsnippet at all.  also would expect "is" or "has" or something for a bool return.  Check doesn't indicate what true or false means as a return value.

Reply:
went with `isBadBuildPhase` in the upcoming push ... added godoc